File: play-java-websocket-example/app/actors/Messages.java
package actors;

import java.util.Set;

import static java.util.Objects.requireNonNull;

public final class Messages {

    public static final class WatchStocks {
        final Set<String> symbols;

        public WatchStocks(Set<String> symbols) {
            this.symbols = requireNonNull(symbols);
        }

        @Override
        public String toString() {
            return "WatchStocks(" + symbols.toString() + ")";
        }
    }

    public static final class UnwatchStocks {
        final Set<String> symbols;

        public UnwatchStocks(Set<String> symbols) {
            this.symbols = requireNonNull(symbols);
        }

        @Override
        public String toString() {
            return "UnwatchStocks(" + symbols.toString() + ")";
        }
    }

}

File: play-scala-websocket-example/app/actors/Messages.scala
package actors

import stocks._

object Messages {

  case class Stocks(stocks: Set[Stock]) {
    require(stocks.nonEmpty, "Must specify at least one stock!")
  }

  case class WatchStocks(symbols: Set[StockSymbol]) {
    require(symbols.nonEmpty, "Must specify at least one symbol!")
  }

  case class UnwatchStocks(symbols: Set[StockSymbol]) {
    require(symbols.nonEmpty, "Must specify at least one symbol!")
  }
}


